### PR TITLE
JS refactor

### PR DESCRIPTION
**Pull request by mfairchild365**
This removes references to `WDN.jQuery()` and improves JS where possible. I also found some bugs along the way which I fixed. There is further room to improve the JS, but this a good start.

### CODE REVIEW

File: src/UNL/MediaHub/Developers/Search.php [hunk 1]
```diff
@@ -5,7 +5,7 @@ class UNL_MediaHub_Developers_Search
     
     public $uri         = "search/{query}";
     
-    public $exampleURI  = "search/idk";
```
Comment (spennythug) [inline src/UNL/MediaHub/Developers/Search.php]: It was funnier this way.

File: www/templates/html/scripts/frontend.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require(['jquery'], function($){
+	WDN.initializePlugin('modal', [function() {
```
Comment (spennythug) [inline www/templates/html/scripts/frontend.js]: I think you can use require to initialize this plugin too. That might be more consistent.

Comment (mfairchild365) [inline www/templates/html/scripts/frontend.js]: Good idea! I took this straight from the examples without much thought. I will see if I can streamline it.

Comment (spennythug) [inline www/templates/html/scripts/frontend.js]: I contributed real feedback!!! 

I think this is how you can do it:
`require(['jquery','modal'], function($,modal){
modal.initialize();
// colorbox stuff
});`

**Pull request merged**